duckduckgo/Android
# Migrate from @OnLifecycleEvent - vpn module

Task/Issue URL: https://app.asana.com/0/0/1202260594569394/f

### Description
@OnLifecycleEvent is now deprecated. Updated code in vpn module to use `DefaultLifecycleObserver`

### Steps to test this PR

- Install from this branch
- Launch the app
- Check in the Logcat that the following Logs appear:

 1. `Registering application lifecycle observer: com.duckduckgo.mobile.android.vpn.blocklist.AppTrackerExclusionListUpdateWorkerScheduler`
 2. `Registering application lifecycle observer: com.duckduckgo.mobile.android.vpn.blocklist.AppTrackerListUpdateWorkerScheduler`
 3.  `Registering application lifecycle observer: com.duckduckgo.mobile.android.vpn.debug.DeviceShieldNotificationsDebugReceiverRegister`
 4.  `Registering application lifecycle observer: com.duckduckgo.mobile.android.vpn.heartbeat.VpnServiceHeartbeatMonitor`
 5.  `Registering application lifecycle observer: com.duckduckgo.mobile.android.vpn.ui.notification.DeviceShieldNotificationScheduler`
